uwblueprint/abtc
# Add UI for creating ServiceRequest

## Notion ticket link

[Add UI for creating ServiceRequest - Include POST request](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d)



## Implementation description
* Create a button that triggers popup for creating new service request (only visible to admins)
* Create a modal popup for creating new service request
* Validate fields on submission
* Send POST request when user submit the form



## Steps to test
1. Create an admin account
2. Log in and navigate to /volunteer-dashboard
<img width="1440" alt="Screenshot 2024-07-23 at 6 52 57 PM" src="https://github.com/user-attachments/assets/fcae18ae-3766-4d0a-a94b-17cf9530aa44">
3. Click on the button to create new service request
<img width="1440" alt="Screenshot 2024-07-23 at 6 53 12 PM" src="https://github.com/user-attachments/assets/1cb4eda9-53c1-4a43-a24f-9c8527d03e95">
<img width="1440" alt="Screenshot 2024-07-23 at 6 53 29 PM" src="https://github.com/user-attachments/assets/5e04d2da-e830-4a7a-9f91-ccb2293cba9b">
<img width="1440" alt="Screenshot 2024-07-23 at 6 53 47 PM" src="https://github.com/user-attachments/assets/83c32210-9ea0-4e48-ba2e-2c9730b6093d">
<img width="1440" alt="Screenshot 2024-07-23 at 6 53 57 PM" src="https://github.com/user-attachments/assets/7aa25826-329e-4697-bd8d-f1df748722a2">
<img width="1440" alt="Screenshot 2024-07-23 at 6 54 07 PM" src="https://github.com/user-attachments/assets/a8eb8ccc-48d2-4a68-8b53-75251be06587">

4. Verify the created shift shows up in the shift cards list view and calendar
5. Create a volunteer account
6. Log in and navigate to /volunteer-dashboard
7. Verify that there's no button to create new service request
<img width="1440" alt="Screenshot 2024-07-23 at 7 56 49 PM" src="https://github.com/user-attachments/assets/0fb4a922-d313-4531-b7dd-78c360e6e841">

Note: if somehow an unknown service request type is passed, user will see this page
<img width="1440" alt="Screenshot 2024-07-23 at 6 52 14 PM" src="https://github.com/user-attachments/assets/03e47e88-485e-4ee8-bef3-9e4cbb256ac3">



## What should reviewers focus on?
* Form validation
  * Errors show if start date is not before end date
  * User should not be able to submit the form with incomplete required fields
* User flow
  * Modal popup closes automatically when the request is successfully created
* Shift requests should have kitchen request fields as null (check in DB)


## Checklist
